### ISSUE TITLE
Publish Documentation with Multiple Environments

### ISSUE THREAD

**Issue by science695** [a85/POSTMan-Chrome-Extension#4818]
I am publishing documentation to a client of mine, and I want to be able to publish both a TESTING and PRODUCTION environment to the client so they have the relvant information about both environments.

When I look at the shared documentation, there is a dropdown that "could" hold more than one environment, but the publish dialog requires me to choose a single environment.

Could you update that publish dialog to allow you to publish multiple environments with a collection?

**Comment by godfrzero** [comment]
We'll see where we can get this added to the roadmap. I'll post updates here as they become available.

**Comment by ghost** [comment]
hey, this is a very important feature, in my work place we buy a licence pro, but we want publish our doc in multiple enviroments. https://api.epayco.co. I hope you fix this soon.

best regards.

**Comment by mcataldo-gr8** [comment]
Any update on this improvement?  This would be valuable for my organization as well.

**Comment by carmenhchung-zz** [comment]
Agreed with all the comments above - if we could have additional environments (we need testing and production, like the original poster) in the Postman documentation, that would be incredibly helpful! :)

**Comment by krishofer** [comment]
+1 on this!

**Comment by vicker** [comment]
+1

**Comment by batshoes** [comment]
+1 this would be great!

**Comment by sankalp0o** [comment]
Hi guys! Thanks for the feedback. Can I suggest you add reactions to the original issue instead of adding new comments? It ends up spamming the OP and others in the thread. Thanks!

So that I understand the problem better, why do you guys need multiple environments? I haven't seen a lot of public APIs with testing environments. Is that assumption incorrect? 
Are your clients looking for multiple environments while going through the documentation or while importing through the Run in Postman button? Is the API public or is it private?

**Comment by carmenhchung-zz** [comment]
Hi @sankalp0o, thanks for getting back to us! 

We offer our partners the option of engaging with both testing and production environments for our public API in Postman through the "Run in Postman" button. As such, we'd like to offer both environments in our Postman documentation as well, so that the root domain can be automatically updated in the documentation to either production or testing, depending on which environment they are interested in using.

Let me know if further info would be helpful!

**Comment by science695** [comment]
@sankalp0o Our use case is similar to what carmenhchung indicated.

We publish the docs for a collection to go to a client beginning with our testing environment, and when they are ready we switch it over to production. It would be much nicer if we could publish both testing and production environments. When I look at our docs (the private un-published versions), there are dropdowns for environments, which seems like it would be a great option to publish several environments and let the documentation be able to switch back and forth.

Perhaps publish multiple environments, and give a default environment option (so the page doesn't load without one).

Thanks!

**Comment by sankalp0o** [comment]
Thanks @carmenhchung and @science695 for providing detailed use cases. This really helps us understand the problem much better.

We will start discussing this and I will let you guys know when we have any updates.

**Comment by radusw** [comment]
My company needs this as well

**Comment by DeanMauro** [comment]
@sankalp0o, are there any updates to share on this feature?

**Comment by mrtibs2000** [comment]
+1

**Comment by mbacon-edocs** [comment]
Slightly different use case, but hopefully same implementation - rather than just two (testing/live) we have the same API on multiple URLs based on the country you're calling from (as well as testing slots in each country). As such we want a single documentation page, and the ability for them to choose the country and then see what the full URL is for them (some of our partners/clients are based in multiple countries as well, so can't just give them 1 pre-built page).

TLDR: Hoping the implementation of this request will allow 2+ environments, not just picking a dev and a live env

**Comment by pratikjain5** [comment]
Hey @sankalp0o @Raja-Simha Not sure what is the status on this issue. Its definitely use case for many companies to provide documentation for Sandbox and Production environments. In our company, we have to provide sandbox environment to our clients and would like to have environment selector on documentation page. It seems we can't publish two environments currently on postman

**Comment by ncri** [comment]
Any update here? We also need this. ;-)

**Comment by shubhpy** [comment]
Any update here in 2021?

**Comment by cdolan92** [comment]
Disappointed this isn't an option. Without it, we have to share our staging baseURL in the documentation body or some other subpar answer.

**Comment by mbraidy** [comment]
This is a very useful feature.
+1 from me
This can serve both the documentation and the sandbox at the same time.
A client can choose one environment to see our server API request url and chose another environment to see the sandbox request url.

**Comment by stewartie4** [comment]
I managed to do this on the docs - I had multiple sets of environments in the dropdown but i can't figure out *how* i did it!

**Comment by stewartie4** [comment]
never mind - because i was logged in, the postman docs ui is rendering my collections alongside the one published with the doc when in actual fact the collection is only published with a single collection - very very deceptive 
can't believe this is not a feature

**Comment by fawcao** [comment]
@sankalp0o any update about this?

**Comment by sankalp0o** [comment]
@Raja-Simha might be able to help here, I'm no longer involved in the project. 

I'd suggest making a public workspace with your collections and multiple environments. This is not the exact thing being asked here, but has a couple of upsides
- You can share not just multiple environments, but multiple collections, mock servers, etc in a single link.
- Public workspaces can help run these requests directly, instead of having to import them first. Any updates are also instantaneous controlled by you.

Hope this helps :)

**Comment by steresi** [comment]
@sankalp0o thanks for the advice here. A Public Workspace would let a client play around with your Collection within the Postman web UI (or I suppose the Postman app also).

However just to confirm... if you create a Public Workspace, say with two Environment Templates ("test" and "production"), that doesn't actually affect the generated documentation page, though, right? The documentation page will still only have one option in the dropdown ("No Environment", plus whatever Environment Template you selected when publishing)!

**Comment by hueldera** [comment]
+1

**Comment by cherryramatisdev** [comment]
+1

**Comment by aharasta** [comment]
Kinda frustrating to not be able to even manually toggle the environments in the app for a set of customers that might be onboarding into a staging environment instead of production. Sharing a public collection is not ideal if we just want to share information publicly without asking customers to familiarize themselves with a new product. @Raja-Simha Where is this on the roadmap?

**Comment by msshahanshah** [comment]
This would be a really helpful feature, can't understand why this has not been implemented since the original request is from 2018!!

**Comment by annoyedcust** [comment]
Even though it's already a dropdown in the UI, I'm beginning to believe we will never see this option made available. 

Want to publish a Prod _and_ a Test environment? No problem. Publish each on a different "custom domain." Why? So they can charge you $10/mo for each extra one.

It would be better if they would just be honest with us. Tell us to buy more domains, and close out the ticket.

**Comment by nickjuntilla** [comment]
Wow this is crazy. This is something that almost all api documentation does. It even looks like it's going to work and then doesn't. How is such a basic feature missing?

**Comment by ianmenethil** [comment]
I can't believe something as basic as this is not available in postman... Sadge

**Comment by lougeo** [comment]
I would also like to see this functionality - there's already a dropdown that can switch between 'No Environment' and the added environment... doesn't seem like a stretch.

**Comment by Ikdemm** [comment]
We would love to have this feature as well. Is there a way we can help you regarding this?

**Comment by MathieuBurnat** [comment]
I still believe. :]

**Comment by raghuchinnannan** [comment]
Is there an update on this? We have 2 different hosts for US & EU data center and we don't want to publish 2  different API documentation. Environment would be ideal.

**Comment by adnenbe** [comment]
Interresting and required feature, any update on this @postman-support  please ?

**Comment by venkatrao225** [comment]
@postman-support We are also running into the same issue, any timelines for the fix?

**Comment by DePalmo** [comment]
The request is four years old (at the time of writing this comment) and this still wasn't added? Wow, just wow.

So next question. Despite I like Postman, this is rather a big issue since I need to post or forward two sets of URLs, which should obviously not be needed: what's the alternative of Postman?

**Comment by venkatrao225** [comment]
@akshaydeo @Raja-Simha could you plz check this conversation and provide us the latest status on this?

**Comment by sharapov-outsource** [comment]
@postman-support do you forget about this? Any update on this? Would be very helpful for us.

**Comment by valentim-guilherme** [comment]
Really? until today not developed?

**Comment by saharc8** [comment]
still not developed! please @postman-staff

**Comment by attack-monkey** [comment]
Would really love to see this developed.

**Comment by julienhauspie** [comment]
Crazy to see that 5 years later this feature is still not present in Postman. @postman-staff @Raja-Simha

**Comment by jayomayan** [comment]
any update on this basically the Environment in Documentation is kinda useless if we can only publish one environment at a time.

**Comment by schengawegga** [comment]
We need this as well. Crazy to see, that there is an dropdown in the published documentation already ;-)

**Comment by richarddwelsh** [comment]
@postman-staff Any update on this? Why have a pull-down menu in the published docs if you can only ever have one environment at a time?

**Comment by mahdizojaji** [comment]
I need this feature too

**Comment by AjatshatruSingh** [comment]
@akshaydeo @Raja-Simha : We need this feature as well. And have been waiting for it for quite a while. Please let us know if this is even in the pipeline anymore. We will plan our usage accordingly.

**Comment by dnandhakumars** [comment]
+1 this would be great!

**Comment by leduyhoang1994** [comment]
It's 2025 already, any update yet ???

**Comment by nickjuntilla** [comment]
7 years for what you would think would be an essential feature. Pretty wild.

**Comment by ethylon** [comment]
Crazy to see this very essential and basic feature is still being ignored after all this time
